lib/ansible/modules/source_control/git.py

```diff
@@ -332,6 +338,24 @@ def write_ssh_wrapper():
     st = os.stat(wrapper_path)
     os.chmod(wrapper_path, st.st_mode | stat.S_IEXEC)
     return wrapper_path
+ 
+def write_key_file(key):
+    '''Write the Git private key to a temporary file'''
+    module_dir = get_module_path()
+    try:
+        # make sure we have full permission to the module_dir, which
+        # may not be the case if we're sudo'ing to a non-root user
+        if os.access(module_dir, os.W_OK|os.R_OK|os.X_OK):
+            fd, key_file = tempfile.mkstemp(prefix=module_dir + '/')
+        else:
+            raise OSError
+    except (IOError, OSError):
+        fd, key_file = tempfile.mkstemp()
+    fh = os.fdopen(fd, 'w+b')
+    fh.write(key)
+    fh.close()
+    os.chmod(key_file, 384) # 0600 octal
```

Comment:
That's required. `0o600` works starting python 2.6. If authorized_key still wants to support 2.4 (not sure when this will die) one can do `int('600', 8)`, but that's probably not more readable than your suggestion.